hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -686,6 +688,38 @@ public void setMaintenanceReplicaMinimum(int replicaCount) {
       this.maintenanceReplicaMinimum = replicaCount;
     }
 
+    /**
+     * Defines how many redundant replicas of a container must be online for a
+     * node to enter maintenance. Currently, only used for EC containers. We
+     * need to consider removing the "maintenance.replica.minimum" setting
+     * and having both Ratis and EC use this new one.
+     */
+    @Config(key = "maintenance.remaining.redundancy",
+        type = ConfigType.INT,
+        defaultValue = "1",
+        tags = {SCM, OZONE},
+        description = "The number of redundant containers in a group which" +
+            " must be available for a node to enter maintenance. If putting" +
+            " a node into maintenance reduces the redundancy below this value" +
+            " , the node will remain in the entering maintenance state until" +
+            " a new replica is created. For Ratis containers, the default" +
+            " value of 1 ensures at least two replicas are online, meaning 1" +
+            " more can be lost without data becoming unavailable. For any EC" +
+            " container it will have at least dataNum + 1 online, allowing" +
+            " the loss of 1 more replica before data becomes unavailable." +
```

Comment:
But for EC, users should aware that, online reconstruction happens, and hence they may see impact in performance. For no performance impact, the very positive case would be [replicaIndex until datanum + 1]. Things will be complicated. I think we must make users aware of the impact when putting nodes in maintenance. 

No plans to add this config in ozone-defaults.xml ?
Do you know this descriptions will be available as part of admin guides or so? I am not sure that happens, then they will be just dev descriptions.

Reply:
The config framework automatically puts these descriptions into the generated ozone-default.xml at build time, thats one of the main reasons for having it.

The point of maintenance mode is to take nodes offline quickly for a short time and avoid the wait and overhead of replication. Of course in any user guide, it should be made clear that maintenance for EC can result in degraded reads. I'm not sure we have such a user guide anywhere to add to!

Reply:
> ozone-default.xml at build time, thats one of the main reasons for having it.

Ok that is good then. 

>The point of maintenance mode is to take nodes offline quickly for a short time and avoid the wait and overhead of replication. Of course in any user guide, it should be made clear that maintenance for EC can result in degraded reads. I'm not sure we have such a user guide anywhere to add to!

May be for now, explicitly mentioning in conf description may be ok. So, that who ever exploring these configs will get attention of the trade offs. Thanks

Reply:
OK - I added a line to the documentation highlighting this.